box/box.go

```diff
@@ -12,11 +12,23 @@ type Box struct {
 
 // New returns a new instance of the box structure, which implements the Box interface.
 func New(conn tarantool.Doer) *Box {
+	if conn == nil {
+		// Check if the provided Tarantool connection is nil, and if it is, panic with an error
+		// message. panic early helps to catch and fix nil pointer issues in the code.
+		panic("tarantool connection cannot be nil")
+	}
```

Comment:
I don't see the point. Whether panic happens here or somewhere else on nil dereference, it will happen anyway and it can be easily identified.

```suggestion
```

Reply:
In user-facing libraries, I think it's better to fail early with a clearer error message than "invalid memory address or nil pointer dereference." While avoiding panics entirely in third-party libraries is ideal, since we don't want to break compatibility, this approach is preferable.